pep/app/pods/admin/route.ts

```diff
@@ -4,11 +4,24 @@ import Route from '@ember/routing/route';
 import { LanguageCode } from 'pep/constants/lang';
 import { PageNav, PageSidebar } from 'pep/mixins/page-layout';
 import AdminController from 'pep/pods/admin/controller';
+import { canAccessRoute } from 'pep/utils/user';
 
 export default class Admin extends PageSidebar(PageNav(Route)) {
     navController = 'admin';
 
-    setupController(controller: AdminController, model: object, transition: Transition) {
+    /**
+     * Check and see if the user can access this route, and if not forward them to the 403 page
+     *
+     * @memberof Admin
+     */
+    beforeModel(): void {
+        const access = canAccessRoute(this, ['user.viewAdmin']);
+        if (!access) {
+            this.transitionTo('four-oh-three');
+        }
```

Comment:
since this use case is so common, i would probably package this up completely into a helper like `authorizedRoute(this, ['ability', ...])` (and use the router service for the transition) so you can just do:

```ts
beforeModel() {
    authorizedRoute(this, ['ability', ...])
}
```